### ISSUE TITLE
Normalization problem

### ISSUE THREAD

**Issue by elbfox** [dams-mcda/Dams-MCDA#133]
So, it turns out the replace NaN operation (lines 261-264 in WSM.R) is too broad of an operation. I'm going through cell by cell in the Ind_NormalizedMatrix and checking to see that the values are correct, because my graphs are still very slightly different. 

I found that EastMill (dam no. 4) Properties  (crit no. 6) normalized values have been replaced with zeros when they should be 1's. It's just that the replace NaN doesn't work in this one particular case. I've tried to solve it like this:

`> Ind_NormalizedMatrix[,6,4]
[1] 0 0 0 0 0
> replace(Ind_NormalizedMatrix[,6,4], c(1:5), c(1,1,1,1,1))`

but the operation doesn't appear to work. I don't think there will be some generalized way to do this because of the NaN issue in the calculation (I have to replace the values manually in my Excel version, for instance), but if you think of one @sythel, please let me know. In the meantime, I am trying to address it in a criteria/dam specific way as above.

**Comment by elbfox** [comment]
replace didn't work but this does:

`Ind_NormalizedMatrix[,6,4] <- c(1,1,1,1,1)`

**Comment by elbfox** [comment]
OK, graphs check out. I'm leaving this as the "fix" for now,  inserted 

`Ind_NormalizedMatrix[,6,4] <- c(1,1,1,1,1)`

at line 265 in WSM.R. **Related problem** (and the reason I'm keeping this issue open) is that the normalization described by @samGroy would replace NaN values with 0 in the normalization for EastMill (Dam4) properties (criteria 6 of 14), when the value should actually be 1. Not sure if the same fix can work here.

**Comment by elbfox** [comment]
Updated (multi-dam) NormalizedMatrix with :

`NormalizedMatrix[4,6,] <- 1 #This should replace the NaN <-- 0 with 0 <-- 1 for East Millinocket`

at line 326, right before the WeightedScoreMatrix is generated.

**Comment by sythel** [comment]
solved in PR #134 ?

**Comment by elbfox** [comment]
No, this actually relates directly to the "rules" from #81. I'm no longer sure that my normalization fix  makes sense, because we would need to take this approach for: 
**number of propertes** at East Millinocket [4,6, ]
**fish habitat** at Medway [2,1, ]
**reservoir storage** at West Enfield [1,3, ], Medway [2,3, ], and East Millinocket [4,3, ]

**Comment by elbfox** [comment]
Also **river recreation** at Millinocket Lake: NormalizedMatrix[7,2, ]

**Comment by elbfox** [comment]
This is what I'm doing locally for Ind_NormalizedMatrix:
```
Ind_NormalizedMatrix[1,6,4] <- 1#This replaces properties NaN at East Millinocket
Ind_NormalizedMatrix[5,1,2] <- 1 #This replaces fish habitat NaN at Medway
Ind_NormalizedMatrix[1,3,1:2] <- 1#This replaces the reservoir storage NaN at West Enfield, Medway
Ind_NormalizedMatrix[1,3,4] <- 1 #This replaces the reservoir storage NaN at East Millinocket
Ind_NormalizedMatrix[5,2,7] <- 1 #This replaces the river rec NaN at Millinocket Lake
```

Because I'm doing it in a decision alternative/criteria/dam specific way here, I'm not actually sure how to approach it for the multi-dam NormalizedMatrix @sythel, so your rules idea would be better, because it's harder to know which decision alternative is happening at any one time.

**Comment by elbfox** [comment]
Updated fix: 
````
	Ind_NormalizedMatrix[2:5,6,3] <- c(1,1,1,1)#This replaces properties NaN at East Millinocket
	Ind_NormalizedMatrix[1,1,2] <- 1 #This replaces fish habitat NaN at Medway
	Ind_NormalizedMatrix[5,3,1:3] <- 1#This replaces the reservoir storage NaN at West Enfield, Medway, East Millinocket
	Ind_NormalizedMatrix[1,2,7] <- 1 #This replaces the river rec NaN at Millinocket Lake
````

**Comment by elbfox** [comment]
@sythel we can close this after the newest branch is merged (to be PR'd later today) and the changes become live.

**Comment by elbfox** [comment]
This issue should be fixed entirely in PR #152 . Closing.
